job/server/src/main/java/alluxio/master/job/plan/PlanTracker.java

```diff
@@ -276,7 +277,7 @@ private synchronized boolean removeFinished() {
   /**
    * @return list of failed plans
    */
-  public List<PlanInfo> failed() {
-    return Lists.newArrayList(mFailed);
+  public Stream<PlanInfo> failed() {
```

Comment:
kind of unusual to pass stream back as return value, is the idea here to save a potential copy?

Reply:
I don't want to send the list back because that's basically making a copy. I don't want to send the underlying data structure back because that makes it accessible. So stream was both the most straightforward and most accessible thing to send back.